packages/strapi-admin/services/permission.js

```diff
@@ -180,6 +189,43 @@ const cleanPermissionInDatabase = async () => {
   }
 };
 
+const ensureBoundPermissionsInDatabase = async () => {
+  if (strapi.EE) return;
+
+  const models = Object.values(strapi.contentTypes);
+  const role = await strapi.query('role', 'admin').findOne({ code: EDITOR_CODE }, []);
+
+  for (const model of models) {
+    const actions = getBoundActionsBySubject(role, model.uid);
+    const perms = await strapi.query('permission', 'admin').find(
+      {
+        subject: model.uid,
+        action_in: actions,
+        role: role.id,
+      },
+      []
+    );
+
+    if (perms.length > 0) {
+      const missingActions = _.difference(actions, _.map(perms, 'action'));
+
+      if (missingActions.length > 0) {
+        const { fields } = perms[0];
```

Comment:
Won't work if the first permission is `DELETE`.
The best is perhaps to do `_.union` for all fields

Reply:
Yup in CE only you can take the first one with fields. you might want to think what happens after a downgrade to CE here. all the permissions might have different fields so there won't be a right one to choose from

Reply:
Let's do a union then